pootle/apps/pootle_config/managers.py

```diff
@@ -0,0 +1,136 @@
+from .exceptions import ConfigurationError
+
+
+class ConfigQuerySet(models.QuerySet):
+
+    def _clone(self, klass=None, setup=None, **kwargs):
+        clone = super(ConfigQuerySet, self)._clone(klass, setup, **kwargs)
+        if getattr(self, "query_model", None):
+            clone.query_model = self.query_model
+        return clone
+
+    def get_config_queryset(self, model):
+        if not model:
+            if getattr(self, "query_model", None):
+                conf = self
+            else:
+                conf = self.site_config()
+        else:
+            conf = self.config_for(model)
+            if not isinstance(model, models.Model):
+                conf = conf.filter(object_pk__isnull=True)
+        return conf
+
+    def get_config(self, key, model=None):
+        return self.get_config_queryset(model).get(key=key).value
+
+    def list_config(self, key=None, model=None):
+        conf = self.get_config_queryset(model)
+        conf_list = []
+
+        if isinstance(key, (list, tuple)):
+            if key:
+                conf = conf.filter(key__in=key)
+        elif key is not None:
+            conf = conf.filter(key=key)
+        for item in conf:
+            # dont use values_list to trigger to_python
+            conf_list.append((item.key, item.value))
+        return conf_list
+
+    def get_model_kwargs(self, model):
+        if model:
+            self.query_model = model
+        elif hasattr(self, "query_model"):
+            model = self.query_model
+        if model is None:
+            return {}
+        ct = ContentType.objects.get_for_model(model)
+        kwargs = dict(content_type=ct)
+        if isinstance(model, models.Model):
+            kwargs["object_pk"] = force_text(model._get_pk_val())
+        return kwargs
+
+    @property
+    def base_qs(self):
+        qs = self.__class__(self.model)
+        if hasattr(self, "query_model"):
+            qs.query_model = self.query_model
+        return qs
+
+    def config_for(self, model):
+        """
+        QuerySet for all comments for a particular model (either an instance or
+        a class).
+        """
+        self.query_model = model
+        return self.base_qs.filter(**self.get_model_kwargs(model))
+
+    def site_config(self):
+        self.query_model = None
+        return self.base_qs.filter(
+            content_type__isnull=True, object_pk__isnull=True)
+
+    def set_config(self, key, value="", model=None):
+        conf = self.get_config_queryset(model)
+        if isinstance(model, models.Model):
+            no_set = config_should_not_be_set.get(
+                model.__class__,
+                instance=model,
+                key=key, value=value)
+        else:
+            no_set = config_should_not_be_set.get(
+                model,
+                key=key, value=value)
+        if no_set:
+            raise ConfigurationError("Unable to save: %s" % no_set)
```

Comment:
im actually gonna get rid of the message altogether, ie
```
ConfigurationError(e)
```
That way it will get the upstream error message properly